include/Token.h

```diff
@@ -197,6 +197,7 @@ enum class KeywordVersion : uint8_t {
 };
 
 const StringTable<TokenKind>* getKeywordTable(KeywordVersion version);
+bool isKeywordOrIdentifier(TokenKind kind);
```

Comment:
Maybe even better would be isPossibleMacroArg or something like that. I'm fine with having a simple isKeyword though; the lexer won't actually produce tokens of those types unless the preprocessor is set properly anyway.